File: sources/com/s8/web/front/carbide/title/H2Title.java
package com.s8.web.front.carbide.title;

import com.s8.io.bohr.neon.core.NeBranch;
import com.s8.io.bohr.neon.core.NeObject;


/**
 * 
 * @author pierreconvert
 *
 */
public class H2Title extends NeObject {
	
	/**
	 * 
	 * @param branch
	 * @param text
	 * @return
	 */
	public H2Title create(NeBranch branch, String text) {
		H2Title title = new H2Title(branch);
		title.setText(text);
		return title;
	}

	
	/**
	 * 
	 * @param branch
	 */
	public H2Title(NeBranch branch) {
		super(branch, "/s8-web-front/carbide/title/H2Title");
	}
	
	
	/**
	 * 
	 * @param name
	 */
	public void setText(String text) {
		vertex.fields().setStringUTF8Field("text", text);
	}
}

File: web-sources/S8-pkgs-ui-carbide/title/H2Title.js

import { NeObject } from '/S8-core-bohr-neon/NeObject.js';

import { S8WebFront } from '/S8-pkgs-ui-carbide/S8WebFront.js';


/**
 * basics
 */
S8WebFront.CSS_import('/S8-pkgs-ui-carbide/title/Title.css');


export class H2Title extends NeObject {

    /**
     * 
     */
    constructor() {
        super();
        this.node = document.createElement("h2");
        this.node.classList.add("title");
    }

    S8_render() { /* continuous rendering approach... */ }

    getEnvelope() {
        return this.node;
    }


    /**
     * 
     * @param {Popover} popoverBox 
     */
    S8_set_text(text) {
        this.node.innerHTML = text;
    }


    S8_dispose() { /* continuous rendering approach... */ }
}